.claude/hooks/session-start.sh

```
@@ -0,0 +1,60 @@
+      <name>Maven Central via repo1 (allowlisted host)</name>
+      <mirrorOf>central</mirrorOf>
+      <url>https://repo1.maven.org/maven2</url>
+    </mirror>
+  </mirrors>
+</settings>
+SETTINGS
+
+# 2) Make the maven-resolver HTTP transport use the JVM proxy system properties
+#    so it routes through the egress proxy instead of connecting directly.
+if [ -n "${CLAUDE_ENV_FILE:-}" ]; then
+  echo 'export MAVEN_OPTS="${MAVEN_OPTS:-} -Daether.connector.http.useSystemProperties=true"' >> "$CLAUDE_ENV_FILE"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)